api_v3/services/SessionService.php

```diff
@@ -299,8 +298,19 @@ function startWidgetSession ( $widgetId , $expiry = 86400 )
 		if(PermissionPeer::isValidForPartner(PermissionName::FEATURE_ENTITLEMENT, $partnerId) &&
 			!is_null($widget->getPrivacyContext()) && $widget->getPrivacyContext() != '' )
 			$privileges .= ','. kSessionBase::PRIVILEGE_PRIVACY_CONTEXT . ':' . $widget->getPrivacyContext();
-		
+
 		$userId = 0;
+
+		// if the widget has a role, pass it in $privileges so it will be embedded in the KS
+		// only if we also have an entry to limit the role operations to
+		if ($widget->getRoles() != null && $widget->getLimitEntry() != null){
```

Comment:
True, but the KMS application won't do that because we write the code and I want to let users to do that because it's their right to do that and because it will open many more applicative flows to use this feature.